jib-plugins-common/src/main/java/com/google/cloud/tools/jib/plugins/common/MainClassResolver.java

```diff
@@ -112,7 +113,13 @@ private static String findMainClassInClassFiles(ProjectProperties projectPropert
                 + "; attempting to infer main class.");
 
     ImmutableList<Path> classesSourceFiles =
-        projectProperties.getJavaLayerConfigurations().getClassesLayerEntry().getSourceFiles();
+        projectProperties
+            .getJavaLayerConfigurations()
+            .getClassLayerEntries()
+            .stream()
+            .peek(layerEntry -> Verify.verify(layerEntry != null))
+            .map(LayerEntry::getSourceFile)
+            .collect(ImmutableList.toImmutableList());
 
     MainClassFinder.Result mainClassFinderResult =
```

Comment:
I'm not a fan of this result-style.  It would make more sense to be to turn the `isSuccess()` into a result-type, and rename `getErrorType()` -> `getResultType()`.  Then your switch would just be the different result types.